app/src/main/java/com/jbnm/homehero/ui/goal/GoalPresenter.java

```diff
@@ -1,58 +1,178 @@
-//        List<Task> tasks = Arrays.asList(task);
-//        reward = new Reward("1", "Disneyland", 20, "castle.jpg");
-//        List<Reward> rewards = Arrays.asList(reward);
-//        child = new Child("1", tasks, rewards);
-        //replace this with code for loading these objects from firebase
+        dataManager = new DataManager();
 
     }
 
     @Override
     public void detach() {
         mvpView = null;
-
     }
 
     @Override
-    public void loadChildData() {
+    public void loadData() {
+        disposable.add(dataManager.getChild("-Kq5YlmM3saCunGh6Jr_")
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeWith(new DisposableObserver<Child>() {
+                    @Override
+                    public void onNext(Child childResult) {
+                        child = childResult;
+                        populateAllTheThings();
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+
+                    }
 
+                    @Override
+                    public void onComplete() {}
+                }));
     }
 
+//    @Override
+    public void populateAllTheThings() {
+        rewards = new ArrayList(child.getRewards().values());
+        mvpView.setGoalDescription(child.currentReward().getDescription());
+        mvpView.setGoalImage(child.currentReward().getRewardImage());
+        checkProgress();
+        determineTaskButtonStatus();
+        mvpView.buildGoalPickerDialog();
+    }
     @Override
     public void checkProgress() {
-        if (child.getTotalPoints() >= reward.getValue()) {
+        if (child.getTotalPoints() >= child.currentReward().getValue()) {
+            mvpView.showProgress(child.currentReward().getDescription(), child.currentReward().getValue(), child.currentReward().getRewardImage(), child.getTotalPoints(), child.calculatePendingPoints());
+            mvpView.showRewardAnimation();
+        } else {
+            mvpView.showProgress(child.currentReward().getDescription(), child.currentReward().getValue(), child.currentReward().getRewardImage(), child.getTotalPoints(), child.calculatePendingPoints());
+        }
+    }
 
+    @Override
+    public void taskButtonClicked() {
+        if (child.currentTask() == null) {
+            mvpView.taskPickerIntent();
         } else {
-            mvpView.showProgress("Disneyland", 20, "sadfasdf", 10, 5);
+            mvpView.taskProgressIntent();
         }
     }
-}
 
+    @Override
+    public void determineTaskButtonStatus() {
+        if (child.allTasksCompleted()) {
+            mvpView.hideTaskButton();
+        } else {
+            mvpView.showTaskButton();
+        }
+    }
+
+    @Override
+    public void setNewRewardAndDecrementPoints(int i) {
+        Log.d(TAG, "total points = " + child.getTotalPoints());
+        child.setTotalPoints(child.getTotalPoints() - child.currentReward().getValue());
```

Comment:
Child has a method redeemReward() that takes care of decrementing points and adds the reward to pending rewards